src/topic.ts

```diff
@@ -377,11 +419,16 @@ export class Topic {
    *   const apiResponse = data[0];
    * });
    */
-  getMetadata(gaxOpts, callback?) {
-    if (is.fn(gaxOpts)) {
-      callback = gaxOpts;
-      gaxOpts = {};
-    }
+  getMetadata(callback: RequestCallback<Topic>): void;
+  getMetadata(gaxOpts: CallOptions, callback: RequestCallback<Topic>): void;
+  getMetadata(gaxOpts?: CallOptions): Promise<Topic>;
+  getMetadata(
+      gaxOptsOrCallback?: CallOptions|RequestCallback<Topic>,
+      callback?: RequestCallback<Topic>): void|Promise<Topic> {
```

Comment:
I believe `google.pubsub.v1.Topic` should be used instead of `Topic` in all cases here.